https://bugzilla.redhat.com/show_bug.cgi?id=617300

LSB and https://fedoraproject.org/wiki/Packaging/SysVInitScript
require status to output something useful, rather than just use
the exit code.

* daemon/libvirt-guests.init.in (status): Break into new routine,
and provide output.
(usage): Document status.
(cherry picked from commit 2c7e5d929c9940ef61103dfa0c8b131bfaa68f78)
---
 daemon/libvirt-guests.init.in |   26 ++++++++++++++++++++------
 1 files changed, 20 insertions(+), 6 deletions(-)

diff --git a/daemon/libvirt-guests.init.in b/daemon/libvirt-guests.init.in
index fdac31d..9ebf184 100644
--- a/daemon/libvirt-guests.init.in
+++ b/daemon/libvirt-guests.init.in
@@ -275,10 +275,28 @@ gueststatus() {
     done
 }

+# rh_status
+# Display current status: whether saved state exists, and whether start
+# has been executed.  We cannot use status() from the functions library,
+# since there is no external daemon process matching this init script.
+rh_status() {
+    if [ -f "$LISTFILE" ]; then
+        echo $"stopped, with saved guests"
+        RETVAL=3
+    else
+        if [ -f "$VAR_SUBSYS_LIBVIRT_GUESTS" ]; then
+            echo $"started"
+        else
+            echo $"stopped, with no saved guests"
+        fi
+        RETVAL=0
+    fi
+}
+
 # usage [val]
 # Display usage string, then exit with VAL (defaults to 2).
 usage() {
-    echo $"Usage: $0 {start|stop|restart|force-reload|gueststatus|shutdown}"
+    echo $"Usage: $0 {start|stop|status|restart|force-reload|gueststatus|shutdown}"
     exit ${1-2}
 }

@@ -299,11 +317,7 @@ case "$1" in
     force-reload)
         ;;
     status)
-        if [ -f "$LISTFILE" ]; then
-            RETVAL=3
-        else
-            RETVAL=0
-        fi
+        rh_status
         ;;
     shutdown)
         ON_SHUTDOWN=shutdown
-- 
1.7.2

